test/e2e/subscription_e2e_test.go

```diff
@@ -467,9 +467,14 @@ var _ = Describe("Subscription", func() {
 		require.NoError(GinkgoT(), err)
 
 		// Wait for the subscription to begin upgrading to csvB
-		subscription, err = fetchSubscription(crc, testNamespace, subscriptionName, subscriptionStateUpgradePendingChecker)
```

Comment:
I wonder if this should be handled at the fetchSubscription layer, and instead, create a new checker type.

/lgtm

Reply:
I think we probably want to start getting rid of a lot of the custom test framework if we can write test specs simply at the call sites.

it seems like maintaining custom test frameworks is one place where the cost of being DRY outweighs its value